packages/components/atoms/f-filter-pill/src/components/FilterPill.vue

```diff
@@ -154,17 +158,29 @@ $filter-pill-checkbox-width: 16px;
     align-self: center;
     opacity: 0;
     width: 0;
+    min-width: 0;
+    transform: translate3d(0, 0, 0) scale(0) rotate(-45deg);
+    will-change: transform, opacity, width, min-width, margin-right;
+
+    @media (prefers-reduced-motion: no-preference) {
+        transition: transform $transition-duration $ease,
+                    opacity $transition-duration $ease,
+                    width $transition-duration $ease,
+                    min-width $transition-duration $ease,
+                    margin-right $transition-duration $ease;
+    }
 
```

Comment:
I wonder if this can be tweaked a bit. `will-change` needs to be used sparingly as its possible it could cause a performance impact rather than improvement.  Also translate3d() is great because it uses the GPU. Could the margin-right be changed using this instead? 

Wondering if you could then transition only - transform, opacity & width and it looks the same 🤞

Reply:
hm. good point! will play a bit more with it

Reply:
I removed `will-change` for now. Let's see if we face any performance issues along the way and if we do we'll see how to refactor it. 

We need `min-width` because otherwise when the text is long, the icon will shrink. 

I don't see a clean readable solution on how to get rid of `margin-right`. If the icon was always visible we could use flex gap. The other solution could be to add `translate3d()` to the text element. But the code will be less maintainable, as the animation part will be spread across different elements. Happy to do it if we see that it's beneficial in terms of performance.